qutebrowser/browser/webengine/spell.py

```diff
@@ -0,0 +1,150 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2017 Michal Siedlaczek <michal.siedlaczek@gmail.com>
+
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Installing and configuring spell-checking for QtWebEngine."""
+
+import os
+from urllib.parse import urljoin
+from urllib.request import urlretrieve
+
+from PyQt5.QtCore import QLibraryInfo
+
+repository_url = 'https://redirector.gvt1.com/edgedl/chrome/dict/'
+
+
+class Language:
+
+    """Dictionary language specs."""
+
+    def __init__(self, code, name, file):
+        self.code = code
+        self.name = name
+        self.file = file
+
+    @staticmethod
+    def from_array(lang_array):
+        """Create Language object from an array.
+
+        Args:
+            lang_array: an array of strings containing
+                        the specs of the language in the following format:
+                        [code, name, file]
+        """
+        return Language.from_tuple(tuple(lang_array))
+
+    @staticmethod
+    def from_tuple(lang_tuple):
+        """Create Language object from a tuple.
+
+        Args:
+            lang_tuple: a tuple of strings containing
+                        the specs of the language in the following format:
+                        (code, name, file)
+        """
+        code, name, file = lang_tuple
+        return Language(code, name, file)
```

Comment:
I don't think this (and `from_array`) is needed, as the calling code can just do `Language(*a_tuple)` or `Language(*a_list)`.